### ISSUE TITLE
Helm chart for kubevirt installation

### ISSUE THREAD

**Issue by vriabyk** [kubevirt/kubevirt#8347]
**Describe the solution you'd like**:
It would be cool to have ability to install kubevirt using helm chart.

**Additional context**:
We are looking for a way to install kubevirt automatically on our kubernetes clusters. We are working on a one-click-deployment for our kubernetes clusters and all addons we need. It would be cool to have ability to deploy kubevirt via helm chart, so we could use for example ArgoCD or terraform to make deployment automated.

Current approach of installing kubevirt via `kubectl apply` is very simple for manual installation, but it isn't good for any automation deployments.

I reviewed old issues and saw concerns like:
```
we can't guarantee it will work with all present/past/future kubevirt releases. The template itself would be specific to a kubevirt release, which then negates the benefit of having a template at all.
```
I think current best practice is to pin helm chart version to release version. Wdyt?

**Comment by gclawes** [comment]
> I think current best practice is to pin helm chart version to release version. Wdyt?

Agreed.  This is what we chose to do in [metallb/metallb), the helm chart is released as a versioned artifact along with binaries/container images.  Software updates that require template changes have them as part of the release.

**Comment by kisahm** [comment]
+1

**Comment by jdambly** [comment]
+1 is there still no chart?

**Comment by fabiand** [comment]
It should be fairly simply to create a helm chart around the kubevirt operator

**Comment by anvarik** [comment]
any updates on this?

**Comment by jdambly** [comment]
> any updates on this?

I've been working on this a little, still needs some work tho
https://github.com/jdambly/kubevirt-charts

**Comment by scydas** [comment]
Any progress?

**Comment by vasiliy-ul** [comment]
@scydas, there is an open proposal in the community repo https://github.com/kubevirt/community/pull/224. Feel free to join the review and provide feedback.

**Comment by PatrickHuetter** [comment]
+1

**Comment by cloudymax** [comment]
I've been working on a set of charts for my homelab using @jdambly work as a starting point, happy to submit them in a PR if there is interest.

https://github.com/cloudymax/kubevirt-community-stack

**Comment by haiwu** [comment]
what's the current status for this? still using kubectl yaml with argocd?

**Comment by tim-van-dijkhuizen** [comment]
+1

**Comment by PatrickHuetter** [comment]
+1

**Comment by kode15333** [comment]
+1

**Comment by alee-x** [comment]
+1

**Comment by xpivarc** [comment]
https://github.com/kubevirt/community/pull/224 is the official proposal on how to implement this. Reviews are welcome.

**Comment by onedr0p** [comment]
Looks like progress is being made https://github.com/kubevirt/charts

/remove-lifecycle stale

**Comment by cade-thomas** [comment]
+1

**Comment by F11GAR0** [comment]
+1

**Comment by joaopedrocg27** [comment]
+1

**Comment by kiemlicz** [comment]
I've really been looking forward to having official Helm charts for KubeVirt, so in the meantime I decided to [generate them automatically](https://github.com/kiemlicz/charter?tab=readme-ov-file#kubevirt).

They’re kept in sync with the [KubeVirt releases](https://github.com/kubevirt/kubevirt/releases), so they should stay up to date automatically.

Feel free to give them a try and share any feedback — it would be great to know if they work well for others too.  
If you find them useful, a "star" the repo please - would mean a lot and help me keep improving it.

**Comment by mohag** [comment]
> I've really been looking forward to having official Helm charts for KubeVirt, so in the meantime I decided to [generate them automatically](https://github.com/kiemlicz/charter?tab=readme-ov-file#kubevirt).
> 
> They’re kept in sync with the [KubeVirt releases](https://github.com/kubevirt/kubevirt/releases), so they should stay up to date automatically.
> 
> Feel free to give them a try and share any feedback — it would be great to know if they work well for others too. If you find them useful, a "star" the repo please - would mean a lot and help me keep improving it.

The images are hardcoded though, which means that changing the registry, which is a common use-case of Helm charts, is not possible. (Ideally the registry, repository and tags should be kept in separate values)

The [recommended ](https://helm.sh/docs/chart_best_practices/labels/#standard-labels) labels and annotations are missing as well.

Many settings are hardcoded and can't be configured. There is no option to add extra volumes / mounts either.

It is useful to make uninstalling kube-virt a bit easier though.

**Comment by kiemlicz** [comment]
@mohag thank you for the feedback

You're right about images and other hardcodes - I've already fixed this for CDI `1.63.2-beta.2` and KubeVirt `1.6.3-beta.1` Chart Versions, please take a look if you have a moment.   

Up till now, my main focus was to have the the mechanism of auto-creating the Chart as KubeVirt is released quite often. Updating their plain manifests was a painful experience

With regards to labels: yes, the helper is already present within the Chart - I'll add its usage too

**Comment by gclawes** [comment]
One way to produce a simpler helm chart for kubevirt is to just produce a chart for the KubeVirt operator.

Tigera does this for `tigera-operator`, the recommended helm chart for installing Calico.  The Calico manifests are somewhat complex to do in a helm chart, but the operator and the CRs for it are relatively simple.

https://github.com/projectcalico/calico/tree/master/charts/tigera-operator

**Comment by kiemlicz** [comment]
> [@mohag](https://github.com/mohag) thank you for the feedback
> 
> You're right about images and other hardcodes - I've already fixed this for CDI `1.63.2-beta.2` and KubeVirt `1.6.3-beta.1` Chart Versions, please take a look if you have a moment.
> 
> Up till now, my main focus was to have the the mechanism of auto-creating the Chart as KubeVirt is released quite often. Updating their plain manifests was a painful experience
> 
> With regards to labels: yes, the helper is already present within the Chart - I'll add its usage too

[CDI 1.64, Kubevirt 1.7.0 contain 'standard' helm labels usage](https://github.com/kiemlicz?tab=packages&repo_name=charter).  
They are automatically added and will be added for next releases to come.    
As of further templating: `.metadata.name` turns to be somewhat tricky to be replaced with template, e.g. Kubevirt `PriorityClass` name can't be changed as further `virt-handler` deployment fails to start (due to use of unchanged name) - dunno I suspect it is hardcoded somewhere in KubeVirt code

As of now I've left Deployment and PriortyClass names unparametrized.
 
Please give it a shot.

**Comment by n-at-han-k** [comment]
For anyone still wondering about this. I set up a repo that auto generates a helm chart with every release of kubevirt. It covers every release from v0.12.0 currently.

https://github.com/general-intelligence-systems/kubevirt-helm

Basically the charts are the equivalent of doing the following steps from the [quickstart guide](https://kubevirt.io/quickstart_cloud/) (so it's just like installing from the docs).

```bash
export VERSION=$(curl -s https://storage.googleapis.com/kubevirt-prow/release/kubevirt/kubevirt/stable.txt)

kubectl create -f "https://github.com/kubevirt/kubevirt/releases/download/${VERSION}/kubevirt-operator.yaml"
kubectl create -f "https://github.com/kubevirt/kubevirt/releases/download/${VERSION}/kubevirt-cr.yaml"
```

**Comment by F11GAR0** [comment]
> For anyone still wondering about this. I set up a repo that auto generates a helm chart with every release of kubevirt. It covers every release from v0.12.0 currently.
> 
> https://github.com/general-intelligence-systems/kubevirt-helm
> 
> Basically the charts are the equivalent of doing the following steps from the [quickstart guide](https://kubevirt.io/quickstart_cloud/) (so it's just like installing from the docs).
> 
> ```bash
> export VERSION=$(curl -s https://storage.googleapis.com/kubevirt-prow/release/kubevirt/kubevirt/stable.txt)
> 
> kubectl create -f "https://github.com/kubevirt/kubevirt/releases/download/${VERSION}/kubevirt-operator.yaml"
> kubectl create -f "https://github.com/kubevirt/kubevirt/releases/download/${VERSION}/kubevirt-cr.yaml"
> ```

looks like a bad way for deploying using operator in helm

the main reason **for me** waiting helm charts is to avoid using of insecure roles for operator

**Comment by n-at-han-k** [comment]
> the main reason **for me** waiting helm charts is to avoid using of insecure roles for operator

feel free to submit a pull request or fork.
